pageserver/src/tenant/vectored_blob_io.rs

```diff
@@ -147,6 +147,21 @@ impl VectoredBlob {
             }
         }
     }
+
+    /// Returns the raw blob, including a 1- or 4-byte header.
+    //
+    // TODO: if buf didn't throw away the header we could just return a full view here.
```

Comment:
@VladLazar I'd be inclined to make this change -- i.e. make `VectoredBlob` also retain the header, and do the parsing during `read()`, so that we can trivially return the existing header + data as a single `BufView` without incurring additional allocations here. Any objections to this?

Reply:
Yeah, that makes sense. Then you don't need the new `write_blob_raw` method.

Reply:
Cool, I'll split out a separate PR for that refactor.

Reply:
Ah, the underlying buffer already contains the header, so this was trivial: #11607.